wallet/src/repository/wallet/mod.rs

```diff
@@ -1061,12 +1075,44 @@ where
             (input_amount - output_amount, model::MovementType::Negative)
         };
 
-        Ok(AccountMutation {
+        // If UTXO set has not changed, then there is no balance movement derived from the transaction being processed
+        if utxo_inserts.is_empty() && utxo_removals.is_empty() {
+            return Ok(None);
```

Comment:
What kinds of transactions can enter this branch?

Reply:
The normal ones, the ones the wallet is indexing :)

Reply:
Is a good question. The wallet is suppose to filter transactions beforehand. It does not harm anyway! :rofl: